iosHyperskillApp/iosHyperskillApp/Sources/Modules/ProjectSelectionList/Views/Grid/Cell/Header/ProjectSelectionListGridCellHeaderView.swift

```diff
@@ -0,0 +1,62 @@
+import SwiftUI
+
+extension ProjectSelectionListGridCellHeaderView {
+    struct Appearance {
+        let spacing = LayoutInsets.defaultInset
+
+        let ratingViewAppearance = StarRatingView.Appearance(
+            imageSize: CGSize(width: 16, height: 16),
+            imageColor: Color(ColorPalette.overlayYellow),
+            textFont: .caption,
+            textColor: .secondaryText
+        )
+    }
+}
+
+struct ProjectSelectionListGridCellHeaderView: View {
+    private(set) var appearance = Appearance()
+
+    let level: SharedProjectLevelWrapper?
+
+    let isGraduate: Bool
+
+    let averageRating: Double
+
+    private var isEmpty: Bool {
```

Comment:
Should we move this logic to shared module?

Reply:
@vladkash In my opinion, part of an architecture that maps some raw data to view data shouldn't know how the view renders those view data, e.g. view mapper just provides some formatted data (don't know about header, list, ...) and the view decides about rendering.

I think it's OK to move this computed property to the shared view mapper, only if we will find some neutral name (not `isHeaderEmpty`).

@XanderZhu what do you think?

Reply:
@ivan-magda 
I agree, that this is a view logic, ViewMapper doesn't know the exact layout. It just provides prepared data to display.